src/test/java/systemtests/TabCommandSystemTest.java

```diff
@@ -0,0 +1,97 @@
+        /* Case: select the first tab in the parcel list, command with leading spaces and trailing spaces
+         * -> tab selected
+         */
+        String command = "   " + TabCommand.COMMAND_WORD + " " + INDEX_FIRST_TAB.getOneBased() + "   ";
+        assertCommandSuccess(command, INDEX_FIRST_TAB);
+
+        /* Case: invalid index (number of tabs + 1) -> rejected */
+        int invalidIndex = TabCommand.NUM_TAB + 1;
+        assertCommandFailure(TabCommand.COMMAND_WORD + " " + invalidIndex, MESSAGE_INVALID_TAB_INDEX);
+
+        /* Case: select the current selected tab -> tab selected */
+        assertCommandSuccess(command, INDEX_FIRST_TAB);
+
+        /* Case: invalid index (0) -> rejected */
+        assertCommandFailure(TabCommand.COMMAND_WORD + " " + 0,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, TabCommand.MESSAGE_USAGE));
+
+        /* Case: invalid index (-1) -> rejected */
+        assertCommandFailure(TabCommand.COMMAND_WORD + " " + -1,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, TabCommand.MESSAGE_USAGE));
+
+        /* Case: invalid arguments (alphabets) -> rejected */
+        assertCommandFailure(TabCommand.COMMAND_WORD + " abc",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, TabCommand.MESSAGE_USAGE));
+
+        /* Case: invalid arguments (extra argument) -> rejected */
+        assertCommandFailure(TabCommand.COMMAND_WORD + " 1 abc",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, TabCommand.MESSAGE_USAGE));
+
+        /* Case: mixed case command word -> rejected */
+        assertCommandFailure("tAB 1", MESSAGE_UNKNOWN_COMMAND);
+
+        /* Case: select from empty address book -> tab selected */
+        executeCommand(ClearCommand.COMMAND_WORD);
+        assert getModel().getAddressBook().getParcelList().size() == 0;
+        assertCommandSuccess(command, INDEX_FIRST_TAB);
+    }
+
+    /**
+     * Executes {@code command} and verifies that the command box displays an empty string, the result display
+     * box displays the success message of executing tab command.
+     * These verifications are done by
+     * {@code AddressBookSystemTest#assertApplicationDisplaysExpected(String, String, Model)}.<br>
+     * Also verifies that the command box has the default style class and the status bar remain unchanged. The resulting
+     * selected tab will be verified if the current selected tab and the tab at
+     * {@code expectedSelectedTabIndex} are different.
+     * @see AddressBookSystemTest#assertApplicationDisplaysExpected(String, String, Model)
+     */
+    private void assertCommandSuccess(String command, Index expectedSelectedTabIndex) {
+        Model expectedModel = getModel();
+        String expectedResultMessage = String.format(
+                MESSAGE_SELECT_TAB_SUCCESS, expectedSelectedTabIndex.getOneBased());
+
+        executeCommand(command);
+        assertApplicationDisplaysExpected("", expectedResultMessage, expectedModel);
+
```

Comment:
create another assert to check that the tab selected and the list shown in the TabPaneHandle is the expected one.